# How does country of citizenship impact certification rates within a couple of states?

## Arizona

In [9]:
#Total number of cases in Arizona, 46
23+18+3+2

46

### Count of Certified, Sorted by country of citizenship

In [ ]:
SELECT case_status, country_of_citizenship, COUNT(country_of_citizenship), job_title_subgroup
FROM salary_data_states_corrected
WHERE work_state='Arizona' AND country_of_citizenship <> 'NA' AND case_status = 'certified'
GROUP BY country_of_citizenship, job_title_subgroup
ORDER BY COUNT(country_of_citizenship) DESC
LIMIT 10

![ac](Images/a_c.png)

In [13]:
#Total Amount of Cases Certified in Arizona
ac = (23/46)*100
ac

50.0

### Count of Certfied-Expired, Sorted by country of citizenship

In [ ]:
SELECT case_status, country_of_citizenship, COUNT(country_of_citizenship), job_title_subgroup
FROM salary_data_states_corrected
WHERE work_state='Arizona' AND country_of_citizenship <> 'NA' AND case_status = 'certified-expired'
GROUP BY country_of_citizenship, job_title_subgroup
ORDER BY COUNT(country_of_citizenship) DESC
LIMIT 10

![ace](Images/a_ce.png)

In [12]:
#Total Amount of Cases Certified-Expired in Arizona
ace = (18/46)*100
ace

39.130434782608695

### Count of Withdrawn, Sorted by country of citizenship

In [ ]:
SELECT case_status, country_of_citizenship, COUNT(country_of_citizenship), job_title_subgroup
FROM salary_data_states_corrected
WHERE work_state='Arizona' AND country_of_citizenship <> 'NA' AND case_status = 'withdrawn'
GROUP BY country_of_citizenship, job_title_subgroup
ORDER BY COUNT(country_of_citizenship) DESC
LIMIT 10

![aw](Images/a_w.png)

In [11]:
#Total Amount of Cases Withdrawn in Arizona
aw = (3/46)*100
aw

6.521739130434782

### Count of Denied, Sorted by country of citizenship

In [ ]:
SELECT case_status, country_of_citizenship, COUNT(country_of_citizenship), job_title_subgroup
FROM salary_data_states_corrected
WHERE work_state='Arizona' AND country_of_citizenship <> 'NA' AND case_status = 'denied'
GROUP BY country_of_citizenship, job_title_subgroup
ORDER BY COUNT(country_of_citizenship) DESC
LIMIT 10

![ad](Images/a_d.png)

In [10]:
#Total Amount of Cases Denied in Arizona
ad = (2/46)*100
ad

4.3478260869565215

## California

In [14]:
#Total Number of Cases in California, 309
140+102+38+29

309

### Count of Certified, Sorted by country of citizenship

In [ ]:
SELECT case_status, country_of_citizenship, COUNT(country_of_citizenship), job_title_subgroup
FROM salary_data_states_corrected
WHERE work_state='California' AND country_of_citizenship <> 'NA' AND case_status = 'certified'
GROUP BY country_of_citizenship, job_title_subgroup
ORDER BY COUNT(country_of_citizenship) DESC
LIMIT 10

![cc](Images/c_c.png)

In [17]:
#Total Amount of Cases Certified in California
cc = (140/309)*100
cc

45.307443365695796

### Count of Certified-Expired, Sorted by country of citizenship

In [ ]:
SELECT case_status, country_of_citizenship, COUNT(country_of_citizenship), job_title_subgroup
FROM salary_data_states_corrected
WHERE work_state='California' AND country_of_citizenship <> 'NA' AND case_status = 'certified-expired'
GROUP BY country_of_citizenship, job_title_subgroup
ORDER BY COUNT(country_of_citizenship) DESC
LIMIT 10

![cce](Images/c_ce.png)

In [18]:
#Total Amount of Cases Certified Expired in California
cce = (102/309)*100
cce

33.00970873786408

### Count of Withdrawn, Sorted by country of citizenship

In [ ]:
SELECT case_status, country_of_citizenship, COUNT(country_of_citizenship), job_title_subgroup
FROM salary_data_states_corrected
WHERE work_state='California' AND country_of_citizenship <> 'NA' AND case_status = 'withdrawn'
GROUP BY country_of_citizenship, job_title_subgroup
ORDER BY COUNT(country_of_citizenship) DESC
LIMIT 10

![cw](Images/c_w.png)

In [20]:
#Total Amount of Cases Withdrawn in California
cw = (38/309)*100
cw

12.297734627831716

### Count of Denied, Sorted by country of citizenship

In [ ]:
SELECT case_status, country_of_citizenship, COUNT(country_of_citizenship), job_title_subgroup
FROM salary_data_states_corrected
WHERE work_state='California' AND country_of_citizenship <> 'NA' AND case_status = 'denied'
GROUP BY country_of_citizenship, job_title_subgroup
ORDER BY COUNT(country_of_citizenship) DESC
LIMIT 10

![cd](Images/c_d.png)

In [19]:
#Total Amount of Cases Denied in California
cd = (29/309)*100
cd

9.385113268608414

## Massachusetts

In [15]:
#Total Number of Cases in Massachusetts, 104
59+33+6+6

104

### Count of Certified, Sorted by country of citizenship

In [ ]:
SELECT case_status, country_of_citizenship, COUNT(country_of_citizenship), job_title_subgroup
FROM salary_data_states_corrected
WHERE work_state='Massachusetts' AND country_of_citizenship <> 'NA' AND case_status = 'certified'
GROUP BY country_of_citizenship, job_title_subgroup
ORDER BY COUNT(country_of_citizenship) DESC
LIMIT 10

![mc](Images/m_c.png)

In [21]:
#Total Amount of Cases Certified in Massachusetts 
mc = (59/104)*100
mc

56.730769230769226

### Count of Certified-Expired, Sorted by country of citizenship

In [ ]:
SELECT case_status, country_of_citizenship, COUNT(country_of_citizenship), job_title_subgroup
FROM salary_data_states_corrected
WHERE work_state='Massachusetts' AND country_of_citizenship <> 'NA' AND case_status = 'certified-expired'
GROUP BY country_of_citizenship, job_title_subgroup
ORDER BY COUNT(country_of_citizenship) DESC
LIMIT 10

![mce](Images/m_ce.png)

In [22]:
#Total Amount of Cases Certified Expired in Massachusetts 
mce = (33/104)*100
mce

31.73076923076923

### Count of Withdrawn, Sorted by country of citizenship

In [ ]:
SELECT case_status, country_of_citizenship, COUNT(country_of_citizenship), job_title_subgroup
FROM salary_data_states_corrected
WHERE work_state='Massachusetts' AND country_of_citizenship <> 'NA' AND case_status = 'withdrawn'
GROUP BY country_of_citizenship, job_title_subgroup
ORDER BY COUNT(country_of_citizenship) DESC
LIMIT 10

![mw](Images/m_w.png)

In [29]:
#Total Amount of Cases Withdrawn in Massachusetts 
mw = (6/104)*100
mw

5.769230769230769

### Count of Denied, Sorted by country of citizenship

In [ ]:
SELECT case_status, country_of_citizenship, COUNT(country_of_citizenship), job_title_subgroup
FROM salary_data_states_corrected
WHERE work_state='Massachusetts' AND country_of_citizenship <> 'NA' AND case_status = 'denied'
GROUP BY country_of_citizenship, job_title_subgroup
ORDER BY COUNT(country_of_citizenship) DESC
LIMIT 10

![md](Images/m_d.png)

In [24]:
#Total Amount of Cases Denied in Massachusetts 
md = (6/104)*100
md

5.769230769230769

## Texas

In [16]:
#Total Number of Cases in Texas, 176
79+56+18+23

176

### Count of Certified, Sorted by country of citizenship

In [ ]:
SELECT case_status, country_of_citizenship, COUNT(country_of_citizenship), job_title_subgroup
FROM salary_data_states_corrected
WHERE work_state='Texas' AND country_of_citizenship <> 'NA' AND case_status = 'certified'
GROUP BY country_of_citizenship, job_title_subgroup
ORDER BY COUNT(country_of_citizenship) DESC
LIMIT 10

![txc](Images/tx_c.png)

In [25]:
#Total Amount of Cases Certified in Texas
tc = (79/176)*100
tc

44.88636363636363

### Count of Certified-Expired, Sorted by country of citizenship

In [ ]:
SELECT case_status, country_of_citizenship, COUNT(country_of_citizenship), job_title_subgroup
FROM salary_data_states_corrected
WHERE work_state='Texas' AND country_of_citizenship <> 'NA' AND case_status = 'certified-expired'
GROUP BY country_of_citizenship, job_title_subgroup
ORDER BY COUNT(country_of_citizenship) DESC
LIMIT 10

![txce](Images/tx_ce.png)

In [26]:
#Total Amount of Cases Certified in Texas
tce = (56/176)*100
tce

31.818181818181817

### Count of Withdrawn, Sorted by country of citizenship

In [ ]:
SELECT case_status, country_of_citizenship, COUNT(country_of_citizenship), job_title_subgroup
FROM salary_data_states_corrected
WHERE work_state='Texas' AND country_of_citizenship <> 'NA' AND case_status = 'withdrawn'
GROUP BY country_of_citizenship, job_title_subgroup
ORDER BY COUNT(country_of_citizenship) DESC
LIMIT 10

![txw](Images/tx_w.png)

In [28]:
#Total Amount of Cases Withdrawn in Texas
tw = (18/176)*100
tw

10.227272727272728

### Count of Denied, Sorted by country of citizenship

In [ ]:
SELECT case_status, country_of_citizenship, COUNT(country_of_citizenship), job_title_subgroup
FROM salary_data_states_corrected
WHERE work_state='Texas' AND country_of_citizenship <> 'NA' AND case_status = 'denied'
GROUP BY country_of_citizenship, job_title_subgroup
ORDER BY COUNT(country_of_citizenship) DESC
LIMIT 10

![td](Images/tx_d.png)

In [27]:
#Total Amount of Cases Denied in Texas
td = (23/176)*100
td

13.068181818181818

# Summary

Note: ('_' indicates no value)

### In Arizona: 
50% of cases were certified, 39.13% were certified-expired, 6.52% were wtihdrawn, and 4.35% were denied.

-most applicants from India and the Philippines

-software engineer from India case status--> 72:22:2:13

-teacher from the Philippines cases staus--> 11:6:8:14

### In California:
45.31% of cases were certified, 33.01% were certified-expired, 12.3% were wtihdrawn, and 9.39% were denied.

-most applicants from India, China, and Canada

-software engineer from India case status--> 1501:603:85:64

-software engineer from China case status--> 408:181:30:9

-software engineer from Canada case status--> 93:47:8:5

### In Massachusetts:
56.73% of cases were certified, 31.73% were certified-expired, 5.77% were wtihdrawn, and 5.77% were denied.

-most applicants from India and China

-software engineer from India case status--> 141:86:12:5

-software engineer from China case status--> 19:12:3:_

### In Texas:
44.89% of cases were certified, 31.82% were certified-expired, 10.23% were wtihdrawn, and 13.07% were denied.

-most applicants from India, Mexico, and the Philippines 

-software engineer from India case status--> 229:152:19:10

-teacher from Mexico case status--> 61:27:5:57

-teacher from the Philippines cases staus--> 13:_:5:20 


